### ISSUE TITLE
Better error message for `map: * xx 2`

### ISSUE THREAD

**Issue by lizmat** [rakudo/rakudo#2729]
$ perl6 -e 'dd <a b c>.map: * x 2'
    ("aa", "bb", "cc").Seq

    $ perl6 -e 'dd <a b c>.map: * xx 2'
    Cannot resolve caller map(List:D: Seq:D); none of these signatures match:
    ($: Hash \h, *%_)
    (\SELF: &block;; :$label, :$item, *%_)
      in block <unit> at -e line 1

Feels to me this difference has a great cognitive dissonance.

Spotted in 
https://stackoverflow.com/questions/54965556/map-signature-mismatch-with-whatever-x-vs-x-vs-xx

**Comment by lizmat** [comment]
Seem Brad Gilbert answered the question: perhaps we need to add some candidates to create a better error message.

**Comment by lizmat** [comment]
FWIW, we already handle the case of passing a `Hash` to a map by an extra candidate, just to create the error message.

    $ perl6 -e 'dd <a b c>.map: { a => 42 }'
    Cannot map a List to a Hash.
    Did you mean to add a stub ({...}) or did you mean to .classify?

Perhaps we should extend that candidate to take all non-Callable values and then adapt the error message accordingly inside the sub?

**Comment by Audiatorix** [comment]
I wonder how many accidents could produce something `Iterable`? It seems like most of them would result from a Whatever not getting curried by one of the exceptions listed in [the doc page](https://docs.perl6.org/type/Whatever).

Perhaps
```perl6
    multi method map(Iterable \i) {
        die "Cannot map a {self.^name} to a {i.^name}.
Did a Whatever get absorbed by a comma, range, series, or list repetition?
Consider using a block if any of these are necessary for your mapping code."
}
```
in `src/core/Any-iterable-methods.pm6`?

**Comment by lizmat** [comment]
@Audiatorix: thanks for the text suggestion.

Implemented with ef8899f619  (with some additions / changes).

Looking at fixing, improving tests now

**Comment by lizmat** [comment]
Tests adapted / added with https://github.com/perl6/roast/commit/88abcf8be8
